src/main/java/org/beanpod/switchboard/controller/DecoderController.java

```diff
@@ -67,10 +73,17 @@
 
   @PutMapping
   public ResponseEntity<DecoderDTO> updateDecoder(@RequestBody DecoderDTO decoderDTO) {
-    Optional<DecoderDTO> decoder = decoderService.findDecoder(decoderDTO.getSerialNumber());
+    Optional<DecoderDTO> decoder = decoderDao.findDecoder(decoderDTO.getSerialNumber());
     if (decoder.isEmpty()) {
       throw new ExceptionType.DeviceNotFoundException(decoderDTO.getSerialNumber());
     }
-    return ResponseEntity.ok(decoderService.save(decoderDTO));
+    return ResponseEntity.ok(decoderDao.save(decoderDTO));
+  }
+
+  @GetMapping("/{serialNumber}/streams")
+  public ResponseEntity<List<StreamModel>> getDecoderStreams(@PathVariable String serialNumber) {
+    List<StreamDTO> streamDtos = decoderService.getDecoderStreams(serialNumber);
+    List<StreamModel> streamModelList = streamMapper.toModel(streamDtos);
+    return ResponseEntity.ok(streamModelList);
```

Comment:
We should wrap this is a catch or use `Optional` to throw a general error if anything wrong happens.

Reply:
The CustomRestExceptionHandler should catch any other errors, but i'll wrap it in Optional to handle any null return values